### DISCUSSION TITLE
Keeping Windows Store Apps Updated with Microsoft Intune - odds+endpoints

### DISCUSSION THREAD

**Discussion by giscus** [ennnbeee/oddsandendpoints-web#1] [Comments]
# Keeping Windows Store Apps Updated with Microsoft Intune

Now we all love the new Windows Store, especially for deploying applications from Microsoft Intune, but we should find a way to keep these UWP applications up to date without additional license cost.

https://memv.ennbee.uk/posts/windows-updating-store-apps/

**Comment by Opportunivore** [comment]
Interestingly this method does NOT work on all APPX packages that need to be updated.  I haven't bothered to dig all the way in but it appears to only effect SOME of the packages that need updating.

**Reply by ennnbeee** [reply]
Have you got some examples of AppxPackages that don't get updated via the Store method? The query is only going to target apps that are Store apps, so any AppxPackage not classed as store app won't be targeted by this remediation script.

**Comment by Opportunivore** [comment]
- Get Help
   - Surface Management Extension
   - Microsoft People
   - Microsoft News
   - MSN Weather
   - Xbox Identity Provider
   - Mixed Reality Portal

There are several others that do not get updated.  I have been working on
this issue on and off for a bit.
Interestingly these packages DO exist in:

Get-CimInstance -Namespace "Root\cimv2\mdm\dmmap" -ClassName
"MDM_EnterpriseModernAppManagement_AppManagement01"

**Reply by ennnbeee** [reply]
Good spot, yeah it looks as though only those AppxPackages that are store registered fall into the ability to update using the provided commands. There looks to be a way to update AppxPackages [directly](https://learn.microsoft.com/en-us/powershell/module/appx/add-appxpackage?view=windowsserver2022-ps#example-2-update-an-app-but-defer-registration-until-the-app-has-closed), so that might be a good place to start if you're looking to update specific apps, or look into the ability to set the [update methods](https://learn.microsoft.com/en-us/powershell/module/appx/set-appxpackageautoupdatesettings?view=windowsserver2022-ps) for all apps.

**Comment by VWSweeny** [comment]
So I am observing that when you psexec and run as system the update method, apps dont actually update.  Nothing happens until a user logs in, then apps are able to update. might be good to note for others trying to figure out why their environment isnt completely up to date.

**Reply by ennnbeee** [reply]
That's an interesting find, sadly there's little documention around the context of the class - https://learn.microsoft.com/en-us/windows/win32/dmwmibridgeprov/mdm-enterprisemodernappmanagement-appmanagement01-updatescanmethod, just that invoking the command will start the update scan method.

How were you observing that the apps weren't updating, were you remote PSEXECing to invoke the command to a device? I could do with testing to see if there is a way around what you've seen.

**Comment by VWSweeny** [comment]
Yes PSEXEX to execute. Then I just watched the apps folder for updates. Nothing updated for several hours.
________________________________

**Reply by ennnbeee** [reply]
OK, so I've tested this and not seen the same behaviour, apps in the C:\Program Files\MicrosoftApps folder are updated whether there is a user signed into the device or not, as long as the device is connected to a network. This is using a vanilla local domain joined WIndows 11 machine, with no GPO/Intune configuration being applied.

**Reply by ennnbeee** [reply]
Soooo further investigation, apps installed on the device in the user context don't get updated unless there's a user signed in. Apps in the system context yes, user context no. 

I'll add an edit to the post this week.

**Comment by Ljalabs** [comment]
Hello,
I'm relatively new to the Intune space for Windows. Real basic question, do apps deployed with Intune from the MS store auto update by default?

What I'm trying to understand is whether or not I need to do this if I'm not blocking the store?

Thanks

**Reply by ennnbeee** [reply]
Yeah they should automatically update, and will if you configure them to using the settings in the post. What the remediation does is kinda force them to update on a regular basis, not just when they feel like it.
